src/zenml/zen_server/routers/steps_endpoints.py

```diff
@@ -244,26 +261,111 @@ def get_step_status(
+        422: error_response,
+    },
 )
 @async_fastapi_endpoint_wrapper
 def get_step_logs(
     step_id: UUID,
-    offset: int = 0,
-    length: int = 1024 * 1024 * 16,  # Default to 16MiB of data
-    strip_timestamp: bool = False,
+    count: int = DEFAULT_PAGE_SIZE,
+    level: int = LoggingLevels.INFO.value,
+    search: Optional[str] = None,
+    page: Optional[int] = None,
+    from_file_index: Optional[int] = None,
+    from_position: Optional[int] = None,
     _: AuthContext = Security(authorize),
-) -> str:
-    """Get the logs of a specific step.
+) -> List[LogEntry]:
+    """Get log entries for efficient pagination.
+
+    This endpoint supports optimized navigation modes:
+
+    - page=1: Returns the latest entries (first page)
+    - page=N without from_position: Requires using page index endpoint first
+    - with from_position: Direct navigation to the page starting from position
+
+    For efficient navigation to arbitrary pages, use the /logs/info endpoint
+    to get page positions, then use those positions with this endpoint.
 
     Args:
         step_id: ID of the step for which to get the logs.
-        offset: The offset from which to start reading.
-        length: The amount of bytes that should be read.
-        strip_timestamp: Whether to strip the timestamp in logs or not.
+        page: The page number to return (1-indexed).
+        count: The number of log entries to return per page.
+        level: Log level filter. Returns messages at this level and above.
+        search: Optional search string. Only returns messages containing this string.
+        from_file_index: Optional file index to start the page from.
+        from_position: Optional position within file to start the page from.
+
+    Returns:
+        List of log entries.
+
+    Raises:
+        HTTPException: If no logs are available for this step or invalid navigation.
+    """
+    step = zen_store().get_run_step(step_id, hydrate=True)
+    pipeline_run = zen_store().get_run(step.pipeline_run_id)
+    verify_permission_for_model(pipeline_run, action=Action.READ)
+
+    store = zen_store()
+
+    # Verify that logs are available for this step
+    if step.logs is None:
+        raise HTTPException(
+            status_code=404, detail="No logs available for this step"
+        )
+
+    # Construct FilePosition if both parameters provided
+    parsed_from_position = None
+    if from_file_index is not None and from_position is not None:
+        parsed_from_position = FilePosition(
+            file_index=from_file_index,
+            position=from_position,
+        )
+
+    try:
+        return fetch_log_records(
+            zen_store=store,
+            artifact_store_id=step.logs.artifact_store_id,
+            logs_uri=step.logs.uri,
+            page=page,
+            count=count,
+            level=level,
+            search=search,
+            from_position=parsed_from_position,
+        )
+    except ValueError as e:
```

Comment:
No need, this happens autoamtically.

Reply:
In general, this is for all endpoints. Don't raise HTTP exceptions, just raise regular exceptions as we already convert them.